### PR TITLE
Created "search-persistent-local-storage" component

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | - |
| Decisions | • This component works like the "search-persistent" component, but the filters are saved to the user session.<br>• Depends on the implementation of "search-persistent" on PR: https://github.com/ripe-tech/ripe-components-vue/pull/166.|
| Animated GIF | - |

### CODE REVIEW

File: vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
+<template>
+    <div class="search-persistent-local-storage">
+        <global />
```
Comment (gcandal) [inline vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue]: Why is this needed?

File: vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.stories.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    v-bind:value.sync="valueData"
+                    v-bind:filters.sync="filtersData"
+                    v-bind:placeholder="placeholder"
+                    v-bind:icon-visible="iconVisible"
+                    v-bind:clear-visible="clearVisible"
+                    v-bind:variant="variant"
+                    v-bind:width="width"
+                    v-bind:loading="loading"
+                    />
+                <p>Value: {{ valueData }}</p>
```
Comment (gcandal) [inline vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.stories.js]: Does this work without emitting anything in the component itself? Do you know why?

File: vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        <search-persistent
+            v-bind:value.sync="valueData"
+            v-bind:filters.sync="filtersData"
+            v-bind:placeholder="placeholder"
+            v-bind:icon-visible="iconVisible"
+            v-bind:clear-visible="clearVisible"
+            v-bind:variant="variant"
+            v-bind:width="width"
+            v-bind:loading="loading"
+            v-on:update:filters="onUpdateFilters"
```
Comment (gcandal) [inline vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue]: Either use `sync` or listen to this event. You could use `sync` and then watch on `filtersData`, for instance.

File: vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        valueData(value) {
+            this.$emit("update:value", this.valueData);
+        },
+        filtersData(value) {
+            this.$emit("update:filters", this.filtersData);
+        }
+    },
+    computed: {
+        filtersKey() {
+            return this.$root.account ? `filters${this.$root.account._id}` : "filters";
```
Comment (gcandal) [inline vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue]: Is `this.$root.account` reactive? Please use the debugger to find out if it's an `Observable`. If it's not then `filtersKey` will need to be a method instead. Remember the `alert` not getting reset bug? It was caused by a similar problem (in that case, the `$refs` not being reactive).

**Comment by gcandal** [comment]
Nice PR 👍

File: vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue (hunk 4 above)
Comment (NFSS10) [reply vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue]: I checked, it is reactive.
Atleast in my knowledge, if it has the `__ob__` property it is reactive.

Comment (gcandal) [reply vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue]: Confirmed: https://github.com/ripe-tech/ripe-pulse/blob/2ce3ade6e91c18aa81b623f15a8db0e192f031c8/js/main.js#L207

File: vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue (hunk 1 above)
Comment (gcandal) [reply vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue]: Ok but that only means that whoever uses this component must include global. If we have a global in every component that defeats the purpose.

Move this to the story instead.

Comment (NFSS10) [reply vue/components/ui/molecules/search-persistent-local-storage/search-persistent-local-storage.vue]: Right, Got it ! 👍

**Comment by gcandal** [comment]
@NFSS10 let's park this until https://github.com/ripe-tech/ripe-components-vue/pull/166 is done. In any case, we'll have to extend this to use a key-by depending on the filter tenancy type, and not only user.

**Comment by joamag** [comment]
Closing with reaper
